src/mcmc/hmc.jl

```diff
@@ -175,6 +214,41 @@ function find_initial_params(
     )
 end
 
+"""
+    initialstep(
+        rng::AbstractRNG,
+        model::AbstractModel,
+        spl::Sampler{<:Hamiltonian},
+        vi::AbstractVarInfo;
+        initial_params=nothing,
+        nadapts=0,
+        verbose::Bool=true,
+        kwargs...
+    )
+
+Perform the initial step for Hamiltonian Monte Carlo sampling.
+
+This function initialises the Hamiltonian, finds a suitable step size (if not provided),
+and performs the first sampling step.
+
+# Keyword Arguments
+
+- `initial_params`: Initial parameter values to use for sampling. If `nothing` (the default),
```

Comment:
`initial_params` and `verbose` work the same way for all samplers, not just HMC, so I would not document them here, but rather refer to the docstring of the more generic `initialstep` function or method. That docstring may need writing too, but would make sense to postpone that after github.com/TuringLang/Turing.jl/pull/2676, because the way `initial_params` works is going to change quite drastically.